assembly/src/ast/nodes/mod.rs

```diff
@@ -287,6 +287,9 @@ pub enum Instruction {
 
     // ----- emit instruction ---------------------------------------------------------------------
     Emit(u32),
+
+    // ----- trace instruction --------------------------------------------------------------------
+    Trace(u32),
```

Comment:
nit: I would put this next to the `Emit` instruction and call the section `event decorators`.